roles/kubernetes-apps/metallb/templates/metallb.yml.j2

```diff
@@ -1860,16 +1871,19 @@ spec:
             - ALL
           readOnlyRootFilesystem: true
       hostNetwork: true
-{% if metallb_speaker_nodeselector %}
       nodeSelector:
-        {{ metallb_speaker_nodeselector | to_nice_yaml | indent(width=8) }}
-{%- endif %}
+        {{ metallb_speaker_nodeselector | to_nice_yaml | indent(width=8) -}}
+        {% if metallb_config.speaker.nodeselector is defined %}
```

Comment:
Maybe here too? This is more OK since it's already behind `metallb_speaker_enabled`, but that defaults to `true` when using MetalLB.